src/box/vy_tx.c

```diff
@@ -370,12 +370,12 @@ vy_tx_is_ro(struct vy_tx *tx)
 static bool
 vy_tx_is_in_read_view(struct vy_tx *tx)
 {
-	return tx->read_view->vlsn != INT64_MAX;
+	return tx->state == VINYL_TX_READY && tx->read_view->vlsn != INT64_MAX;
```

Comment:
AFAICS everywhere this helper is called, `tx->state` is `VINYL_TX_READY` so this extra check seems redundant.

Reply:
Fixed: removed this check, as we discussed.